feat: add image picker component

jefrydco opened the pull request:
## Description

This PR add image picker component. There are 2 ways of using the component.

The first one is **wrapping uninteractive element as a child**, whenever the element is pressed, it will trigger image picker.

```tsx
<ImagePicker
  launchType="camera"
  preview={imagePickerResponse =>
    imagePickerResponse && (
      <Preview imagePickerResponse={imagePickerResponse} />
    )
  }
  options={{
    mediaType: 'photo',
  }}>
  <Text style={style.text}>Pick as Text</Text>
</ImagePicker>
```

The next one is **wrapping interactive as a child function**, it receives a parameter `openPicker` that can be called to trigger image picker:

```tsx
<ImagePicker
  launchType="camera"
  preview={imagePickerResponse =>
    imagePickerResponse && (
      <Preview imagePickerResponse={imagePickerResponse} />
    )
  }
  options={{
    mediaType: 'photo',
  }}>
  {({openPicker}) => (
    <View style={style.button}>
      <Button onPress={openPicker} title="Pick as Button" />
    </View>
  )}
</ImagePicker>
```
## Loom
https://user-images.githubusercontent.com/20434351/150628966-987d203d-d4f8-4d3e-b237-4ddc557c4166.mp4

## Issue

Close: https://github.com/Jitera/jitera-frontend/issues/505

zrg-team commented on src/components/atoms/ImagePicker/Component.tsx:
```diff
@@ -0,0 +1,54 @@
+    | ((childrenParam: { openPicker: OpenPicker }) => void);
+}
+
+const ImagePicker = forwardRef<View, ImagePickerProps>(
```
can we use another library with Compatible with web? like expo-image (can use it over bare project over unimodule, it implemeted already in generated project)

zrg-team commented on src/components/atoms/ImagePicker/Component.tsx:
```diff
@@ -0,0 +1,54 @@
+      React.useState<ImagePickerResponse>();
+
+    const openPicker = React.useCallback<OpenPicker>(async () => {
+      if (launchType === 'camera') {
```
should we have permission implemented for it ?

zrg-team commented:
@jefrydco it should use default render input if user don't want to customize. 
image picker should have props: (for form implement)
+ onChange trigger when selected a image
+ value show when user already choose a image
+ Error message to show error validation in this field

jefrydco commented:
@sieuhuflit @zrg-team I've fixed the image picker, please kindly help recheck. I also fixed several bugs:

- [x] onPress callback on Button isn't triggered
- [x] Can't render icon component 

https://user-images.githubusercontent.com/20434351/151185088-f04d4ab0-c03c-48fb-a42c-caefec931352.mp4

zrg-team commented:
@jefrydco please remove react-native-image-picker if we don't use it
